src/openfl/display/DisplayObjectContainer.hx

```diff
@@ -676,6 +676,38 @@ class DisplayObjectContainer extends InteractiveObject {
 	}
 	
 	
+	override function __cleanup ():Void {
+		
+		super.__cleanup ();
+		
+		for (child in __children) {
+			
+			child.__cleanup ();
+			
+		}
+		
+		__cleanupRemovedChildren ();
+		
+	}
+	
+	
+	inline function __cleanupRemovedChildren () {
+		
+		for (orphan in __removedChildren) {
+			
+			if (orphan.stage == null) {
+				
+				orphan.__cleanup ();
+				
+			}
+			
+		}
+		
+		__removedChildren.length = 0;
```

Comment:
As also mentioned in https://github.com/innogames/openfl/pull/57#discussion_r203802882:

I have a gut feeling that using simple array for `__removedChildren` and just setting it to an empty one here would perform better than this whole `Vector` machinery, but this is a bit off-topic for this PR.